aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -3496,6 +3496,18 @@ Tensor _weight_int4pack_mm_cpu(
   return C;
 }
 
+Tensor _weight_int4pack_mm_cpu_tensor(
+    const Tensor& A,
+    const Tensor& B,
+    const Tensor& qGroupSize,
+    const Tensor& qScaleAndZeros) {
+  TORCH_CHECK(qGroupSize.numel() == 1, __func__, ": qGroupSize should be a scalar tensor.");
+  TORCH_CHECK(qGroupSize.scalar_type() == at::kLong,
```

Comment:
Can you please add an assert for the groupsize's value being correct?
It must be equal to `(B.numel() * 8)/(qScalesAndZeros.numel() / 2)`.

Thanks!